metagraph/src/graph/representation/hash/dbg_sshash.cpp

```diff
@@ -99,6 +100,35 @@ void DBGSSHash::add_sequence(std::string_view sequence,
     throw std::logic_error("adding sequences not supported");
 }
 
+template <class Dict, bool with_rc>
+void map_to_nodes_with_rc_advanced(size_t k,
+                                   const Dict &dict,
+                                   std::string_view sequence,
+                                   const std::function<void(sshash::lookup_result)>& callback,
+                                   const std::function<bool()>& terminate) {
+    size_t n = sequence.size();
+    if (terminate() || n < k)
+        return;
```

Comment:
If we have this check here, it's probably fine to remove it from `map_to_nodes_with_rc`. And we  should probably also move `if (!num_nodes())` check from `map_to_nodes_with_rc` here as well.